php-checks/src/test/resources/checks/LoopExecutingAtMostOnceCheck.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+
+function foo($i)
+{
+    while ($i < 10) { // Noncompliant
+//  ^^^^^
+        echo $i;
+        $i++;
+        break;
+    }
+
+    while ($i < 10) { // Noncompliant
+//  ^^^^^
+        if ($i == 5) {
+            break;
+//          ^^^^^^< {{loop exit}}
+        } else {
+            return;
+//          ^^^^^^^< {{loop exit}}
+        }
+    }
+
+    while ($i < 10) { // Noncompliant
+        echo $i;
+        $i++;
+        throw $error;
+    }
+
+    while ($i < 10) { // Noncompliant
+        echo $i;
+        $i++;
+        goto label;
```

Comment:
`label` is not declared, how we build this CFG?

Reply:
it seems to be handled by creating ad-hoc block https://github.com/SonarSource/sonar-php/blob/95bd6cf42e078e4a7a993db701fd2fc2646d6583/php-frontend/src/main/java/org/sonar/php/cfg/ControlFlowGraphBuilder.java#L267